examples/reverse_words/__init__.py

```diff
@@ -220,24 +252,60 @@ def main(mode, save_path, num_batches, data_path=None):
                                   save_separately=["model", "log"]),
                 Printing(every_n_batches=1)])
         main_loop.run()
-    elif mode == "test":
-        logger.info("Model is loaded")
-        chars = tensor.lmatrix("features")
-        generated = generator.generate(
-            n_steps=3 * chars.shape[0], batch_size=chars.shape[1],
-            attended=encoder.apply(
-                **dict_union(fork.apply(lookup.lookup(chars),
-                             as_dict=True))),
-            attended_mask=tensor.ones(chars.shape))
+    elif mode == "sample" or mode == "beam_search":
+        chars = tensor.lmatrix("input")
+        generated = reverser.generate(chars)
         model = Model(generated)
+        logger.info("Loading the model..")
         model.set_param_values(load_parameter_values(save_path))
-        sample_function = model.get_theano_function()
-        logging.info("Sampling function is compiled")
+
+        def generate(input_):
```

Comment:
Is it better to define it in the global scope? It is non-pickable and users who are going to study this example may have problems.